crates/engine/tree/src/tree/mod.rs

```diff
@@ -571,6 +571,10 @@ where
             }
         };
 
+        let header = block.clone_sealed_header();
```

Comment:
instead of cloning the full header here, can we just send the `BlockNumHash` in `BlockReceived` variant?